src/Slider/Slider.jsx

```diff
@@ -576,9 +626,10 @@ const Slider = class Slider extends React.Component {
 
     // slider tray
     const trayStyle = {};
-    const trans = pct(slideSize * currentSlide * -1);
+    const additionalSlides = infinite ? 1 : 0;
+    const trans = pct((slideSize * currentSlide + additionalSlides * slideSize) * -1);
 
-    if (this.state.isBeingTouchDragged || this.state.isBeingMouseDragged || disableAnimation) {
+    if (this.state.isBeingTouchDragged || this.state.isBeingMouseDragged || this.state.disableAnimationTemp || disableAnimation) {
```

Comment:
Would we be able to leverage the existing `disableAnimation`? Using the temp may have side effects if other places are using the `disableAnimation` property.